### PR TITLE
Mark interest

### PR DESCRIPTION
**Pull request by yongsokheng**

### CODE REVIEW

File: app/views/mark_interests/_mark_interest.html.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<%= link_to "javascript:void(0)", class: "mark-interest-action btn btn-default",
+  data: {marked_id: marked_id} do %>
+  <i class="fa fa-heart fa-red" aria-hidden="true"></i>&nbsp;
+  <%= t("profiles.actions.mark_interest") %>
```
Comment (NgengChhengkim) [inline app/views/mark_interests/_mark_interest.html.erb]: bo ()

File: app/views/mark_interests/_unmark_interest.html.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<%= link_to "javascript:void(0)", class: "unmark-interest-action btn btn-default",
+  data: {mark_interest_id: mark_interest_id} do %>
+  <i class="fa fa-heart-o fa-red" aria-hidden="true"></i>&nbsp;
+  <%= t("profiles.actions.unmark_interest") %>
```
Comment (NgengChhengkim) [inline app/views/mark_interests/_unmark_interest.html.erb]: bo ()

File: app/models/user.rb [hunk 3]
```diff
@@ -42,6 +42,8 @@ class User < ApplicationRecord
   has_many :save_posts, dependent: :destroy
   has_many :like_posts, dependent: :destroy
   has_many :share_posts, dependent: :destroy
+  has_many :mark_interests, class_name: MarkInterest, foreign_key: :marker_id, dependent: :destroy
+  has_many :marked_interests, class_name: MarkInterest, foreign_key: :marked_id, dependent: :destroy
```
Comment (NgengChhengkim) [inline app/models/user.rb:46]: co can quan he nay khong?

File: app/views/mark_interests/create.js.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<% if @mark_interest.errors.any? %>
+  $(".flash").html("<%= j render 'layouts/flash' %>");
+<% else %>
+  $(".mark-interest-actions").html("<%= j render 'mark_interests/unmark_interest', mark_interest_id: @mark_interest.id %>");
+<% end %>
+
```
Comment (NgengChhengkim) [inline app/views/mark_interests/create.js.erb]: bo

File: app/models/user.rb (hunk 3 above)
Comment (yongsokheng) [reply app/models/user.rb:46]: ot brer te,tae dak oy hoy mong

**Pull request merged**